indico/web/flask/util.py

```diff
@@ -326,6 +329,24 @@ def to_url(self, value):
         return super().to_url(value)
 
 
+class SignedValueConverter(BaseConverter):
+    def __init__(self, map, /, salt, max_age=None):
+        super().__init__(map)
+        self.salt = salt
+        self.max_age = max_age
+        self.serializer = secure_serializer if max_age else static_secure_serializer
+
+    def to_python(self, value):
+        try:
+            return self.serializer.loads(value, salt=self.salt, max_age=self.max_age)
+        except BadData:
+            raise NotFound(_('Invalid or expired signature'))
+
+    def to_url(self, value):
+        signed = self.serializer.dumps(value, salt=self.salt)
```

Comment:
Just so we don't forget about it:

Allowing to generate signed URLs unconditionally in here is a risky, because we have e.g. the RHSignURL endpoint which lets you generate a url for an arbitrary endpoint and params. This is not a problem in general, but combined with this converter here, it would of course let someone get the correctly-signed string for an endpoint.

My suggestion would be to add a new `_sign=False` kwarg to our custom `url_for`, which sets something on `g` before calling the Flask `url_for`, and then adding a check in here that bails out with an exception if that param is not set. I would also add a unit test that ensures `RHSignURL` indeed does not allow passing any underscored args to `url_for`, and that it won't build a URL containing a signed param.